cmd/invoker/main.go

```diff
@@ -0,0 +1,228 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/lambda"
+)
+
+type LambdaEvent struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+	URL      string `json:"url"`
+	Debug    bool   `json:"debug"`
+	Reload   bool   `json:"reload"`
+	Delay    int    `json:"delay"`
+	JoinTeam string `json:"join_team,omitempty"`
+}
+
+type LambdaResponse struct {
+	StatusCode int               `json:"statusCode"`
+	Headers    map[string]string `json:"headers"`
+	Body       string            `json:"body"`
+}
+
+type ResponseBody struct {
+	Message    string `json:"message"`
+	Screenshot string `json:"screenshot,omitempty"`
+	Error      string `json:"error,omitempty"`
+}
+
+func main() {
+	// Define command line flags
+	username := flag.String("username", "", "Username for login (with number suffix after '-')")
+	password := flag.String("password", "", "Password for login")
+	url := flag.String("url", "", "URL of the Mattermost instance")
+	debug := flag.Bool("debug", false, "Enable debug mode (includes screenshot)")
+	reload := flag.Bool("reload", false, "Enable reload after delay")
+	delay := flag.Int("delay", 5000, "Delay in milliseconds after login")
+	joinTeam := flag.String("join_team", "", "Team name to join")
+	functionName := flag.String("function", "mattermost-login", "Name of the Lambda function")
+	region := flag.String("region", "us-east-1", "AWS region")
+	outputFile := flag.String("output", "", "File to save screenshot (if debug is enabled)")
+	count := flag.Int("count", 1, "Number of concurrent Lambda invocations")
+
+	flag.Parse()
+
+	// Validate required parameters
+	if *username == "" || *password == "" || *url == "" {
+		fmt.Println("Error: username, password, and url are required")
+		flag.Usage()
+		return
```

Comment:
We should ideally use `os.Exit` with a non-zero code.

Reply:
Thanks, I looked into it. By default, it uses `ExitOnError` (`var CommandLine = NewFlagSet(os.Args[0], ExitOnError)`). And ExitOnError is defined as `Call os.Exit(2) or for -h/-help Exit(0).` So it seems like exit(2) is the idiomatic usage for any flag error.